src/main/java/uk/gov/justice/probation/courtcaseservice/service/OffenderService.java

```diff
@@ -31,40 +31,40 @@ public OffenderService(final OffenderRestClient client, final DocumentTypeFilter
 
-        Tuple3<Offender, List<Conviction>, GroupedDocuments> tuple3 = Mono.zip(client.getOffenderByCrn(crn), client.getConvictionsByCrn(crn), client.getDocumentsByCrn(crn))
+        Tuple3<ProbationRecord, List<Conviction>, GroupedDocuments> tuple3 = Mono.zip(client.getOffenderByCrn(crn), client.getConvictionsByCrn(crn), client.getDocumentsByCrn(crn))
             .blockOptional()
             .orElseThrow(() -> new OffenderNotFoundException(crn));
 
-        Offender offender = combineOffenderAndConvictions(tuple3.getT1(), tuple3.getT2());
-        combineConvictionsAndDocuments(offender, tuple3.getT3().getConvictions(), applyFilter);
+        ProbationRecord probationRecord = combineOffenderAndConvictions(tuple3.getT1(), tuple3.getT2());
+        combineConvictionsAndDocuments(probationRecord, tuple3.getT3().getConvictions(), applyDocumentFilter);
 
-        return offender;
+        return probationRecord;
     }
 
-    private void combineConvictionsAndDocuments(final Offender offender, final List<ConvictionDocuments> convictionDocuments, boolean applyFilter) {
+    private void combineConvictionsAndDocuments(final ProbationRecord probationRecord, final List<ConvictionDocuments> convictionDocuments, boolean applyDocumentFilter) {
 
         final ConcurrentMap<String, List<OffenderDocumentDetail>> allConvictionDocuments;
         allConvictionDocuments = convictionDocuments.stream()
                                 .map(convictionDocument -> ConvictionDocuments.builder()
                                                                 .convictionId(convictionDocument.getConvictionId())
                                                                 .documents(convictionDocument.getDocuments().stream()
-                                                                            .filter(doc -> (!applyFilter || documentTypeFilter.test(doc)))
+                                                                            .filter(doc -> (!applyDocumentFilter || documentTypeFilter.test(doc)))
                                                                             .collect(Collectors.toList()))
                                                                 .build())
                                 .collect(Collectors.toConcurrentMap(ConvictionDocuments::getConvictionId, ConvictionDocuments::getDocuments));
 
-        offender.getConvictions()
+        probationRecord.getConvictions()
             .forEach((conviction) -> {
                 final String convictionId = conviction.getConvictionId();
                 conviction.setDocuments(allConvictionDocuments.getOrDefault(convictionId, Collections.emptyList()));
             });
     }
 
-    private Offender combineOffenderAndConvictions(Offender offender, List<Conviction> convictions) {
-        offender.setConvictions(convictions);
-        return offender;
+    private ProbationRecord combineOffenderAndConvictions(ProbationRecord probationRecord, List<Conviction> convictions) {
```

Comment:
maybe rename to addConvictionsToProbationRecord or similar?

Reply:
OK -